u_parrots.go

```diff
@@ -2076,11 +2076,11 @@ func (uconn *UConn) ApplyPreset(p *ClientHelloSpec) error {
 }
 
 func (uconn *UConn) generateRandomizedSpec() (ClientHelloSpec, error) {
-	return generateRandomizedSpec(uconn.ClientHelloID, uconn.serverName, uconn.HandshakeState.Session, uconn.config.NextProtos)
+	return generateRandomizedSpec(&uconn.ClientHelloID, uconn.serverName, uconn.HandshakeState.Session, uconn.config.NextProtos)
 }
 
 func generateRandomizedSpec(
-	id ClientHelloID,
+	id *ClientHelloID,
```

Comment:
Here.

Reply:
Using pointers here instead may indeed improve some performance, but I wouldn't say it is a bug. 

UTLS is not (yet) optimized for performance anyways... It is better be treated as a prototype instead of product/service. 

And, good work!

Reply:
It's for https://github.com/refraction-networking/utls#reusing-randomized-fingerprint